common/reducers/swap.js

```diff
@@ -92,8 +74,10 @@ export function swap(state: State = INITIAL_STATE, action: SwapAction) {
       };
     }
     case 'SWAP_DESTINATION_KIND': {
+      const newOriginKind = buildOriginKind(state.originKind, action.value);
```

Comment:
Previously, we were assuming that there would never be an action created in which the destination kind was the same as the origin kind.